src/ol/source/ImageArcGISRest.js

```diff
@@ -141,8 +147,9 @@ class ImageArcGISRest extends ImageSource {
     if (this.url_ === undefined) {
       return null;
     }
-    if (!this.loader) {
+    if (!this.loader || this.loaderProjection_ !== projection) {
```

Comment:
The `equivalent()` function should be used here

Reply:
The code replicates the behavior before version 8 - if the view projection was EPSG:3857 the image was requested for CRS EPSG:3857, if the view projection was changed to EPSG:900913 the image was requested for CRS EPSG:900913.  Ditto for EPSG:4326/CRS:84, etc.